embulk-standards/src/main/java/org/embulk/standards/JsonParserPlugin.java

```diff
@@ -99,12 +131,79 @@ private PageBuilder newPageBuilder(Schema schema, PageOutput output)
         return new PageBuilder(Exec.getBufferAllocator(), schema, output);
     }
 
-    private JsonParser.Stream newJsonStream(FileInputInputStream in)
-            throws IOException
-    {
-        return new JsonParser().open(in);
+    private JsonParser.Stream newJsonStream(FileInputInputStream in , PluginTask task)
+            throws IOException {
+        InvalidEscapeStringPolicy policy = task.getInvalidEscapeStringPolicy();
+        switch (policy) {
+            case SKIP:
+            case UNESCAPE:
+                Iterable<CharSource> lines = Lists.transform(CharStreams.readLines(new BufferedReader(new InputStreamReader(in))),
+                        invalidEscapeStringFunction(policy));
+                return new JsonParser().open(new ReaderInputStream(CharSource.concat(lines).openStream()));
+            case PASSTHROUGH:
+            default:
+                return new JsonParser().open(in);
+        }
+    }
+
+    Function<String, CharSource> invalidEscapeStringFunction(final InvalidEscapeStringPolicy policy) {
+        return new Function<String, CharSource>() {
+            final Pattern p = Pattern.compile("\\p{XDigit}+");
+            @Override
+            public CharSource apply(@Nullable String input) {
+                assert input != null;
+                if (policy == InvalidEscapeStringPolicy.PASSTHROUGH) {
+                    return CharSource.wrap(input);
+                }
+                StringBuilder s = new StringBuilder();
+                char[] charArray = input.toCharArray();
+                for (int i=0 ; i < charArray.length ; i++) {
+                    char c = charArray[i];
+                    if (c == '\\') {
+                        if (charArray.length > i + 1) {
+                            char next = charArray[i + 1];
+                            switch (next) {
+                                case 'b':
+                                case 'f':
+                                case 'n':
+                                case 'r':
+                                case 't':
+                                case '"':
+                                case '\\':
+                                case '/':
+                                    s.append(c);
+                                    break;
+                                case 'u': // hexstring such as \u0001
+                                    if (charArray.length > i + 5) {
```

Comment:
@smdmts Thanks. It's perfect for `UNESCAPE`, but it may be worth considering `SKIP`.

Only from `SKIP`'s simple definition like "remove these invalid escapes", behavior for `\\u` may not be trivial. For example, it can be still natural expectation for some users that `\\u12` is removed from `\\u12xY` or `\\u12<EOL>`.

Reasonable expectations for `\\u...` on `SKIP` may be:
1. Only `\\u` is always removed.
    * `\\u1234` ==> `1234` / `\\u12xY` ==> `12xY`
2. Whole `\\uXXXX` is removed if valid. Only `\\` is removed if invalid. (This PR)
    * `\\u1234` ==> (empty) / `\\u12xY` ==> `u12xY`
3. Whole `\\uXXXX` is removed if valid. Only `\\u` is removed if invalid. 
    * `\\u1234` ==> (empty) / `\\u12xY` ==> `12xY`
4. Whole `\\uX...` is removed even if the length of its valid part is shorter than 4.
    * `\\u1234` ==> (empty) / `\\u12xY` ==> `xY`

I thought 2 is not very consistent with `SKIP`'s definition to be honest. What do you think about 1 or 4?